tests/integration/component.go

```diff
@@ -309,4 +309,93 @@ func componentTests(args ...string) {
 		})
 	})
 
+	Context("odo component updating", func() {
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+			context = helper.CreateNewContext()
+		})
+
+		JustAfterEach(func() {
+			helper.DeleteProject(project)
+			os.RemoveAll(context)
+		})
+
+		It("should be able to update a component from local to git", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", append(args, "create", "nodejs", "cmp-git", "--project", project, "--min-memory", "100Mi", "--max-memory", "300Mi", "--min-cpu", "0.1", "--max-cpu", "2", "--context", context, "--app", "testing")...)
+			helper.CmdShouldPass("odo", "push", "--context", context, "-v", "4")
+			getMemoryLimit := oc.MaxMemory("cmp-git", "testing", project)
+			Expect(getMemoryLimit).To(ContainSubstring("300Mi"))
+			getMemoryRequest := oc.MinMemory("cmp-git", "testing", project)
+			Expect(getMemoryRequest).To(ContainSubstring("100Mi"))
+			getCPULimit := oc.MaxCPU("cmp-git", "testing", project)
+			Expect(getCPULimit).To(ContainSubstring("2"))
+			getCPURequest := oc.MinCPU("cmp-git", "testing", project)
+			Expect(getCPURequest).To(ContainSubstring("100m"))
+
+			// update the component config according to the git component
+			helper.CmdShouldPass("odo", "config", "set", "sourcelocation", "https://github.com/openshift/nodejs-ex", "--context", context, "-f")
+			helper.CmdShouldPass("odo", "config", "set", "sourcetype", "git", "--context", context, "-f")
+
+			// check if the earlier resource requests are still valid
+			helper.CmdShouldPass("odo", "push", "--context", context, "-v", "4")
+			getMemoryLimit = oc.MaxMemory("cmp-git", "testing", project)
+			Expect(getMemoryLimit).To(ContainSubstring("300Mi"))
+			getMemoryRequest = oc.MinMemory("cmp-git", "testing", project)
```

Comment:
In the next push only one validation would be enough. 

Unnecessary till line no 349. So remove line no 344 to 349